src/indexes.ts

```diff
@@ -197,9 +196,7 @@ class Index<T = Record<string, any>> {
     const req = new HttpRequests(config)
     const task = await req.post(url, { ...options, uid })
 
-    task.enqueuedAt = new Date(task.enqueuedAt)
-
-    return task
+    return new EnqueuedTask(task)
```

Comment:
The `@returns` line 188 is outdated. Could you update it with 
```
* @returns {Promise<EnqueuedTask>}
```

Reply:
Sure, resolved this.